lib/languages/thingtalk/constants.js

```diff
@@ -39,7 +39,7 @@ function extractConstants(ast) {
 
     ast.visit(new class extends Ast.NodeVisitor {
         visitStringValue(value) {
-            addConstant('QUOTED_STRING', value.value, value);
+            addConstant('QUOTED_STRING', value.value.replace(/[ \t\v\r\n]+/g, ' ').trim(), value);
```

Comment:
why is this a problem? don't we want to at least preserve the linebreaks?

Reply:
This is used to make the output sentence from the agent. The string will be in the middle of some larger sentence. Visually,  a line break looks out of place, and of course, there is no line break in speech...